src/main/java/roshow/RoController.java

```diff
@@ -0,0 +1,83 @@
+package roshow;
+
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import roshow.storage.StorageFileNotFoundException;
+import roshow.storage.StorageService;
+
+
+@Controller
+public class RoController {
+
+    private final StorageService storageService;
+
+    @Autowired
+    public RoController(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    @GetMapping("/")
+    public String listUploadedFiles(Model model) throws IOException {
+
+        model.addAttribute("files", storageService.loadAll().map(
+                path -> MvcUriComponentsBuilder.fromMethodName(RoController.class,
+                        "serveFile", path.getFileName().toString()).build().toString())
+                .collect(Collectors.toList()));
+
+        return "uploadForm";
+    }
+
+    @GetMapping("/files/{filename:.+}")
```

Comment:
OK for first go, but for later you will need to include some kind of UUID or similar in the path `@GetMapping`s, so that multiple people can upload different ROs at the same time.

Even without a database you can just use the same UUID as part of the path of where you save the ROs.